daily_milestone.go

```diff
@@ -158,34 +158,57 @@ func getMilestones(baseURL string, token string, projectID string) ([]simpleMile
 func createMilestoneData(advance int, timeInterval string) []simpleMilestone {
 	today := time.Now().Local()
 	list := []simpleMilestone{}
-	for i := 0; i < advance; i++ {
-		if timeInterval == "daily" {
+
+	if timeInterval == "daily" {
+		for i := 0; i < advance; i++ {
 			date := today.AddDate(0, 0, i).Format("2006-01-02")
 			milestone := simpleMilestone{}
 			milestone.Title = date
 			milestone.DueDate = date
 			list = append(list, milestone)
-		} else if timeInterval == "weekly" {
-			today = today.AddDate(0, 0, 7)
-			year, week := today.ISOWeek()
-			milestoneYear := strconv.Itoa(year)
-			milestoneWeek := strconv.Itoa(week)
+		}
+	} else if timeInterval == "weekly" {
+		newDate := today
+		day := newDate.Weekday()
+		year, week := newDate.ISOWeek()
+		x := day - day // initializing x to zero in time.Time
+		for i := 0; i < advance; i++ {
+			for j := x; j < 7; j++ {
+				z := day - day + 1
+				if z+day > 6 {
+					day = 0
+					newDate = newDate.AddDate(0, 0, 1)
+					break
+
+				} else if day == 0 {
+					day = 0
+					break
+				} else {
+					day = day + z
+					newDate = newDate.AddDate(0, 0, 1)
+				}
+			}
+			newDate = newDate.AddDate(0, 0, 1)
+			day = newDate.Weekday()
+			year, week = newDate.ISOWeek()
 			milestone := simpleMilestone{}
-			milestone.Title = milestoneYear + "-w" + milestoneWeek
+			milestone.Title = strconv.Itoa(year) + "-w" + strconv.Itoa(week)
 			milestone.DueDate = today.Format("2006-01-02")
```

Comment:
This doesn't actually set the right DueDate as you are using today and not newDate, which I assume you would want.